### PR TITLE
Include autotag script as generic ROCm tool

### PR DESCRIPTION
**Pull request by lawruble13**

### CODE REVIEW

File: tools/Autotag/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Autotag
+
+Use of this tagging scripts requires the project to have a changelog as described in the confluence link below
+
+https://confluence.amd.com/display/MLSE/Changelogs+and+Release+Notes
```
Comment (saadrahim) [inline tools/Autotag/README.md:5]: That's an internal link. Let's make the changelog requirements public. I can handle that separately.

Comment (saadrahim) [reply tools/Autotag/README.md:5]: Let's not merge until the changelog guidelines are public.

File: tools/Autotag/tag_script.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    return args
+
+
+def run_tagging():
+    """Run the tagging/releasing process on each specified library."""
+    args = parse_args()
+
+    if args.manifest_url is None:
+        manifest_url = (
+            f"http://rocm-ci.amd.com/job/compute-rocm-rel-{args.version}"
```
Comment (saadrahim) [inline tools/Autotag/tag_script.py:189]: change to default.xml in root directory of this repo

**Review by saadrahim — APPROVED** [review]

File: tools/autotag/tag_script.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        "rocmswplatform": "ROCmSoftwarePlatform",
+        "rocmswplatform-internal": "ROCmSoftwarePlatform",
+        "amdgpuopenlib": "GPUOpen-ProfessionalCompute-Libraries",
+        "radeonopencompute": "RadeonOpenCompute",
+        "radeonopencompute-internal": "RadeonOpenCompute"
+    }
+
+    libraries: list[ReleaseLib] = []
+
+    for entry in manifest_tree.findall(".*[@groups='mathlibs']"):
```
Comment (Naraenda) [inline tools/autotag/tag_script.py:222]: I'm rebasing my PR onto this to merge the scripts. This won't work with how the manifest `default.xml` is setup in this reposotiry. Is it OK to add this property?

**Pull request merged**